config/v1/types_ingress.go

```diff
@@ -41,13 +44,107 @@ type IngressSpec struct {
+	Name string `json:"name"`
+
+	// hostname is the hostname that should be supported by the route.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Format=hostname
+	// +kubebuilder:validation:Required
+	// +required
+	Hostname string `json:"hostname"`
+
+	// servingCertKeyPairSecret is a reference to a secret in namespace/openshift-config that is a kubernetes tls secret.
+	// The serving cert/key pair must match and will be used by the operator to fulfill the intent of serving with this name.
+	// +kubebuilder:validation:Required
+	// +required
+	ServingCertKeyPairSecret SecretNameReference `json:"servingCertKeyPairSecret"`
+}
+
+// ComponentRouteStatus contains information allowing configuration of a route's hostname and serving certificate.
+type ComponentRouteStatus struct {
+	// namespace is the namespace of the route to customize. It must be a real namespace. Using an actual namespace
+	// ensures that no two components will conflict and the same component can be installed multiple times.
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +kubebuilder:validation:Required
+	// +required
+	Namespace string `json:"namespace"`
+
+	// name is the logical name of the route to customize. It does not have to be the actual name of a route resource.
+	// Keep in mind that this is your API for users to customize. You could later rename the route, but you cannot rename
+	// this name.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// defaultHostname is the normal hostname of this route.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Format=hostname
+	// +kubebuilder:validation:Required
+	// +required
+	DefaultHostname string `json:"defaultHostname"`
+
+	// consumingUsers is a slice of ServiceAccounts that need to have read permission on the ServingCertKeyPairSecret secret.
+	// +optional
+	ConsumingUsers []string `json:"consumingUsers"`
+
+	// currentHostnames is the list of current names used by the route.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:Format=hostname
+	// +kubebuilder:validation:Required
+	// +required
+	CurrentHostnames []string `json:"currentHostnames"`
+
+	// configConditions are used to communicate the state of a ComponentRoute.
+	//
+	// Supported conditions include degraded and progressing.
+	// If Degraded is true that means something has gone wrong trying to handle the ComponentRoute.
+	// The CurrentHostnames may or may not be operating successfully.
+	//
+	// If Progressing is true that means the component is taking some action related to the ComponentRoute.
+	// +optional
+	Conditions []ClusterOperatorStatusCondition `json:"configConditions,omitempty"`
 
+	// relatedObjects are a list of resources which are useful when debugging or inspecting how this is applied.
+	// +optional
+	RelatedObjects []corev1.ObjectReference `json:"relatedObjects,omitempty"`
```

Comment:
There should always be at least one item in this list.

Reply:
Understood, I will mark the field as required and add a minimum length.